src/pybind/mgr/dashboard/frontend/src/app/shared/api/ceph-service.service.ts

```diff
@@ -39,6 +39,20 @@ export class CephServiceService {
     );
   }
 
+  update(serviceSpec: { [key: string]: any }) {
+    const serviceName = serviceSpec['service_id']
+      ? `${serviceSpec['service_type']}.${serviceSpec['service_id']}`
+      : serviceSpec['service_type'];
+    return this.http.put(
+      `${this.url}/${serviceName}`,
+      {
+        service_name: serviceName,
+        service_spec: serviceSpec
+      },
```

Comment:
I guess this would require a model or a new class definition?

Reply:
Editing a service is pretty much similar to the service creation (both require service name and service spec as parameters), I don't think this would require a separate model to be introduced.